BAZProjectC1/Components/CarosuelMenu/CarosuelMenu.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  CarosuelMenu.swift
+//  BAZProjectC1
+//
+//  Created by 1044336 on 23/09/22.
+//
+
+import UIKit
+
+struct CarosuelMenuConfiguration {
+    let frame: CGRect
+    let optionsTitles: [String]
+    let itemBackgroundColor: UIColor
+    let itemBorderBackgroundColor: UIColor
+    let itemSelectedBackgroundColor: UIColor
+    let itemSelectedBorderBackgroundColor: UIColor
+    let itemBorder: Double
+    
+    init(frame:CGRect,
+         optionsTitles:[String] = [],
+         itemBackgroundColor: UIColor = .clear,
+         itemBorderBackgroundColor: UIColor = .clear,
+         itemSelectedBackgroundColor: UIColor = .clear,
+         itemSelectedBorderBackgroundColor: UIColor = .clear,
+         itemBorder: Double = 10.0) {
+        
+        self.frame = frame
+        self.optionsTitles = optionsTitles
+        self.itemBackgroundColor = itemBackgroundColor
+        self.itemBorderBackgroundColor = itemBorderBackgroundColor
+        self.itemSelectedBackgroundColor = itemSelectedBackgroundColor
+        self.itemSelectedBorderBackgroundColor = itemSelectedBorderBackgroundColor
+        self.itemBorder = itemBorder
+    }
+}
+
+class CarosuelMenu: UIView {
+    public weak var delegate: CarosuelMenuDelegate?
```

Comment:
¿Es necesario que este delegate sea público? ¿Podría ser internal?

Tienes el mismo typo aquí:

```suggestion
    weak var delegate: CarouselMenuDelegate?
```